spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/TraceFilterTests.java

```diff
@@ -386,6 +392,9 @@ public void closesSpanWhenResponseStatusIs2xxAndExceptionIsClientAbortException(
 				.header(Span.TRACE_ID_NAME, 20L).buildRequest(new MockServletContext());
 		TraceFilter filter = new TraceFilter(beanFactory());
 		BDDMockito.given(beanFactory.getBean(ErrorController.class)).willReturn(() -> "/error");
+		List<ExceptionToIgnoreInTraceFilter> filters = Lists.newArrayList(getClientAbortExpcetionToIgnoreInTraceFilter());
```

Comment:
I think we should add one test with a filter that throws the `ClientAbortException` and we would like to ensure that the span got closed and reported. WDYT?

Reply:
like this?
``` java
@Test
public void closesSpanWhenResponseStatusIs2xxAndClientAbortExceptionThrowAfterTraceFilter() throws Exception {
	this.request = builder().header(Span.SPAN_ID_NAME, PARENT_ID)
			.header(Span.TRACE_ID_NAME, 20L).buildRequest(new MockServletContext());
	TraceFilter filter = new TraceFilter(beanFactory());
	BDDMockito.given(beanFactory.getBean(ErrorController.class)).willReturn(() -> "/error");
	List<ExceptionToIgnoreInTraceFilter> filters = Lists.newArrayList(getClientAbortExpcetionToIgnoreInTraceFilter());
	BDDMockito.given(beanFactory.getBean(ExceptionToIgnoreInTraceFilterProvider.class))
			.willReturn(getExceptionToIgnoreInTraceFilterProvider(filters));
	this.response = new MockHttpServletResponse();
	response.setStatus(200);
	this.filterChain = new MockFilterChain(){
		@Override
		public void doFilter(ServletRequest request, ServletResponse response) throws IOException, ServletException {
			throw new ClientAbortException();
		}
	};
	try {
		filter.doFilter(this.request, this.response, this.filterChain);
	}catch (ClientAbortException e){
		// ig
	}
	then(TestSpanContextHolder.getCurrentSpan()).isNull();
}
```

Reply:
yeah and also

```java
	then(TestSpanContextHolder.getCurrentSpan()).isNull();
         // verify that the span got reported
         then(spanReporter.getSpans()).hasSize(1);
```